gnd/src/main/java/com/google/android/gnd/persistence/remote/RemoteStorageManager.java

```diff
@@ -31,5 +32,5 @@
   Task<Uri> getDownloadUrl(String remoteDestinationPath);
 
   /** Uploads file to a remote path. */
-  Completable uploadMediaFromFile(File file, String remoteDestinationPath);
+  Flowable<Event<UploadProgress>> uploadMediaFromFile(File file, String remoteDestinationPath);
```

Comment:
Why wrapping with `Event`? It's used as a "Wrapper for events passed through streams that should be handled at most once.". It's primarily used to prevent UIs from showing the same message more than ones. In this case, it seems you'd want to just stream the UploadProgress with `Backpressure.LATEST`.